TestingHelpers/MockFileStreamFactory.cs

```diff
@@ -0,0 +1,116 @@
+using System.Security.AccessControl;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Abstractions.TestingHelpers
+{
+    [Serializable]
+    public class MockFileStreamFactory : IFileStreamFactory
+    {
+        private readonly IMockFileDataAccessor mockFileSystem;
+
+        public MockFileStreamFactory(IMockFileDataAccessor mockFileSystem)
+        {
+            if (mockFileSystem == null)
+            {
+                throw new ArgumentNullException("mockFileSystem");
+            }
+
+            this.mockFileSystem = mockFileSystem;
+        }
+
+        public Stream Create(string path, FileMode mode)
+        {
+            return new MockFileStream(mockFileSystem, path, GetStreamType(mode));
+        }
+
+        public Stream Create(string path, FileMode mode, FileAccess access)
+        {
+            return new MockFileStream(mockFileSystem, path, GetStreamType(mode, access));
+        }
+
+        public Stream Create(string path, FileMode mode, FileAccess access, FileShare share)
+        {
+            return new MockFileStream(mockFileSystem, path, GetStreamType(mode, access));
+        }
+
+        public Stream Create(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize)
+        {
+            return new MockFileStream(mockFileSystem, path, GetStreamType(mode, access));
+        }
+
+        public Stream Create(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options)
+        {
+            return new MockFileStream(mockFileSystem, path, GetStreamType(mode, access));
+        }
+
+        public Stream Create(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, bool useAsync)
+        {
+            return new MockFileStream(mockFileSystem, path, GetStreamType(mode, access));
+        }
+
+        public Stream Create(string path, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
```

Comment:
The interface has a preprocessor check, but this is exposed

Reply:
Oops, will add a check here.